shipstation/api.py

```diff
@@ -398,3 +418,23 @@ def post(self, endpoint='', data=None):
         )
         if self.debug:
             pprint.PrettyPrinter(indent=4).pprint(r.json())
+
+    def fetch_orders(self, parameters_dict=None):
+        invalid_parameters = []
+
+        if parameters_dict is None:
+            parameters_dict = {}
+        else:
+            invalid_parameters = [
+                key for key in parameters_dict.keys() if key not in self.ORDER_LIST_PARAMETERS
+            ]
+
+        if not invalid_parameters:
```

Comment:
Try to avoid negative conditions unless they're mandatory (e.g., guard clause suggestion above)

```
if invalid_parameters:
    raise AttributeError("...")
```

Additionally, since you're `raise`ing out of the method if one of the params isn't valid, there's not really any reason to extract the "good" params, because the method execution will never reach completion if any fail.

I think it's reasonable that if bad params are passed in, we simply fail out. It should prevent unexpected results.

Reply:
That makes sense. I didn't think about `raise` like the end statement it is.